src/drawers/drawer.ts

```diff
@@ -106,9 +168,20 @@ export class Drawer {
         selection.attr(colorAttribute, step.attrToAppliedProjector[colorAttribute]);
       }
     });
-    step.animator.animate(selection, step.attrToAppliedProjector);
+    // this is a temporary workaround to ensure that any existing custom animators
+    // (using the 'legacy' api point 'animate') continue to work as before
+    // this test should be removed if and when semver = 2.0.0
+    if (step.animator.animateJoin) {
+      step.animator.animateJoin(this._joinResult, step.attrToAppliedProjector, this);
+    } else {
+      // an old animator won't remove exit
+      this._joinResult.exit
+        .remove();
+      step.animator.animate(selection, step.attrToAppliedProjector);
+    }
```

Comment:
good morning @bluong 
I have thought further on the issue of the change to animator interface, specifically adding and using the ```animateJoin``` method as in here.

First point to note is that for the majority of cases, the javascript consumer never calls the ```animator``` directly, it is called on their behalf by the ```Drawer``` when they assign the animator to their plot.

So, the only case that presents a problem is if a javascript client has built a custom animator and implemented ```animate``` - ```animate``` would not be called, and in fact there would be an error at this point because ```animateJoin``` is not found.

This (unattractive) change will defend against this.

I think the only circumstance now where any consumer of plottable will see a difference is that if you are authoring a custom animator in typescript, you'll get a compile warning about the missing member.